n3fit/src/n3fit/model_trainer.py

```diff
@@ -343,7 +350,7 @@ def _xgrid_generation(self):
             op.split, op_args=sp_ar, op_kwargs=sp_kw, name="pdf_split"
         )
 
-        return InputInfo(input_layer, sp_layer, inputs_idx)
+        return InputInfo(input_layer, sp_layer, inputs_idx), inputs_unique
```

Comment:
I don't really like passing the same information twice, pick a format and stick with it. Maybe you can reconstruct inputs_unique after calling the xgrid generation

Reply:
Yes.

`xinput.input_l.tensor_content` contains the array.

Btw, is no called `input_l` in master because following the review suggestion the name was changed. Please rebase to current master.